File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # approach: two arrays of char counts, compare
        
        if len(s) != len(t):
            return False
        
        s_counts, t_counts = [0] * 26, [0] * 26
        
        for i in range(len(s)):
            s_counts[ord(s[i]) - ord('a')] += 1
            t_counts[ord(t[i]) - ord('a')] += 1
        
        return s_counts == t_counts

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if (s.size() != t.size()) {
            return false;
        }
        
        vector<int> counts_s(26, 0);
        vector<int> counts_t(26, 0);
        
        for (int i = 0; i < s.size(); i++) {
            counts_s[s[i] - 'a']++;
            counts_t[t[i] - 'a']++;
        }
        
        return counts_s == counts_t;
    }
};